dandiapi/api/datacite.py

```diff
@@ -0,0 +1,246 @@
+"""
+DataCite API client implementation.
+
+This module provides the implementation details for interacting with the DataCite API.
+The public interface is exposed through doi.py.
+"""
+
+from __future__ import annotations
+
+import copy
+import logging
+from typing import TYPE_CHECKING
+
+from django.conf import settings
+import requests
+
+if TYPE_CHECKING:
+    from dandiapi.api.models import Version
+
+# All of the required DOI configuration settings
+# Cannot be in doi.py to avoid circular imports
+DANDI_DOI_SETTINGS = [
+    (settings.DANDI_DOI_API_URL, 'DANDI_DOI_API_URL'),
+    (settings.DANDI_DOI_API_USER, 'DANDI_DOI_API_USER'),
+    (settings.DANDI_DOI_API_PASSWORD, 'DANDI_DOI_API_PASSWORD'),
+    (settings.DANDI_DOI_API_PREFIX, 'DANDI_DOI_API_PREFIX'),
+]
+
+logger = logging.getLogger(__name__)
+
+
+class DataCiteClient:
+    """Client for interacting with the DataCite API."""
+
+    def __init__(self):
+        self.api_url = settings.DANDI_DOI_API_URL
+        self.api_user = settings.DANDI_DOI_API_USER
+        self.api_password = settings.DANDI_DOI_API_PASSWORD
+        self.api_prefix = settings.DANDI_DOI_API_PREFIX
```

Comment:
Previously, we would use `settings.DANDI_DOI_API_PREFIX or '10.80507'`

If api prefix is not set, DOI API operations should be prevented by `is_configured()` so I think this is appropriate